### PR TITLE
[WIP] Port unique users to use Analytics server

### LINKED ISSUE #224: Replace number of users to come from Cube.js

### PR DESCRIPTION
**Pull request by dalmia**
Fixes #224 

## Summary
- [x] replace unique users to come from Plio Analytics server instead of backend
- [x] Updated environment variables for the Plio Analytics server
- [x] Updated `docs/ENV.md` to add explanations for new env variables
- [x] make this work with staging
- [ ] make this work with production

## Test Plan
Tested locally
Test on Staging - TODO

### CODE REVIEW

File: src/services/API/RootClient.js [hunk 1]
```diff
@@ -6,18 +6,25 @@ import {
...
 };
 
+// backend API client
 const client = axios.create({
   baseURL: process.env.VUE_APP_BACKEND,
   withCredentials: false,
   headers,
 });
 
+// analytics API
```
Comment (deepansh96) [inline src/services/API/RootClient.js:23]:
```suggestion
// analytics API client
```

File: src/services/API/Plio.js [hunk 2]
```diff
@@ -85,14 +90,30 @@ export default {
...
+    var uniqueUsersQuery = {
+      measures: ["Session.uniqueUsers"],
+      filters: [
+        {
+          member: "Plio.uuid",
+          operator: "equals",
+          values: [plioId],
+        },
+      ],
+    };
```
Comment (deepansh96) [inline src/services/API/Plio.js:111]: I think it's best to encapsulate this query and all the queries that we will use soon in a separate file and import them here. Just like we've done for endpoints and API calls.
A query directory can have different files like Users, Session, Plios etc and each file will have the related query in a variable or const

Comment (deepansh96) [inline src/services/API/Plio.js:111]: Before this cube js change, we were querying the plio ka endpoint to get this metric. But here we're querying sessions ka cube, why is that? can't it be done by querying the plio ka cube?

**Review by deepansh96 — CHANGES_REQUESTED** [review]

File: docs/ENV.md [hunk 3]
```diff
@@ -12,6 +12,13 @@ Token related URL at backend repo which can be used to convert the OAuth2 token
 #### `VUE_APP_BACKEND_WEBSOCKET`
 Base URL to connect to any configured websocket listeners on the backend
 
+### Plio analytics
+#### `VUE_APP_CUBEJS_AUTH_TOKEN`
+Auth token for the analytics REST API - can be empty for local development.
```
[the remark this answers is not in the document]
Comment (dalmia) [reply docs/ENV.md:17]: Yes

File: src/services/API/Endpoints.js [hunk 4]
```diff
@@ -11,4 +11,6 @@ export const pliosEndpoint = "/plios/",
   usersEndpoint = "/users/",
   userConfigEndpoint = "/config/",
   convertTokenEndpoint = "/convert-token/",
-  refreshTokenEndpoint = "/token/";
+  refreshTokenEndpoint = "/token/",
+  listPliosEndpoint = "list_uuid/",
```
[the remark this answers is not in the document]
Comment (dalmia) [reply src/services/API/Endpoints.js:15]: Not how it was being used earlier.

File: src/services/API/Queries/Plio.js [hunk 5]
```diff
@@ -0,0 +1,15 @@
+export function uniqueUsersQuery(plioId) {
+  // query to fetch the number of unique users for the given plio ID
+  // this is the query in the CubeJS Data Schema which is automatically
+  // converted to SQL by CubeJS: https://cube.dev/docs/getting-started-cubejs-schema
```
Comment (deepansh96) [inline src/services/API/Queries/Plio.js:4]: Do we really need this?
`this is the query in the CubeJS Data Schema which is automatically converted to SQL by CubeJS`
Can keep the link for reference purposes

File: src/services/API/Plio.js [hunk 6]
```diff
@@ -8,6 +8,7 @@ import {
   listPliosEndpoint,
   duplicatePlioEndpoint,
 } from "@/services/API/Endpoints.js";
+import { uniqueUsersQuery } from "./Queries/Plio.js";
```
Comment (deepansh96) [inline src/services/API/Plio.js:11]: `import { uniqueUsersQuery } from "@/services/API/Queries/Plio.js";`

File: src/services/API/Analytics.js [hunk 7]
```diff
@@ -0,0 +1,11 @@
+import { apiClient } from "@/services/API/RootClient.js";
+
+export default {
+  async getAnalyticsAccessToken() {
+    return apiClient().post(
+      "/cubejs-token",
```
Comment (dalmia) [inline src/services/API/Analytics.js:6]: should be in the endpoints file

File: src/store/modules/auth.js [hunk 8]
```diff
@@ -97,6 +104,9 @@ const mutations = {
   updateUserStatus(state, status) {
     state.user.status = status;
   },
+  setAnalyticsAccessToken(state, accessToken) {
```
Comment (dalmia) [inline src/store/modules/auth.js:107]: don't we need an unset method that we might need when we have to refresh the token?

File: src/services/API/Analytics.js [hunk 9]
```diff
@@ -0,0 +1,11 @@
+import { apiClient } from "@/services/API/RootClient.js";
+
+export default {
+  async getAnalyticsAccessToken() {
```
Comment (dalmia) [inline src/services/API/Analytics.js:4]: one line comment.

**Pull request merged**